spec/unit/typesafe/config/conf_parser_spec.rb

```diff
@@ -59,3 +63,760 @@ def parse(s)
     end
   end
 end
+
+def parse_path(s)
+  first_excepton = nil
+  second_exception = nil
+  # parser first by wrapping into a whole document and using the regular parser
+  result =
+      begin
+        tree = parse_without_resolving("[${#{s}}]")
+        if tree.is_a?(Hocon::ConfigList)
+          ref = tree[0]
+          if ref.is_a?(Hocon::Impl::ConfigReference)
+            ref.expression.path
+          end
+        end
+      rescue Hocon::ConfigError => e
+        first_excepton = e
+        nil
+      end
+
+  # also parse with the standalone path parser and be sure the outcome is the same
+  begin
+    should_be_same = Hocon::Impl::PathParser.parse_path(s)
+    unless result == should_be_same
+      raise ScriptError, "expected '#{result}' to equal '#{should_be_same}'"
+    end
+  rescue Hocon::ConfigError => e
+    second_exception = e
+  end
+
+  if first_excepton.nil? && (!second_exception.nil?)
+    raise ScriptError, "only the standalone path parser threw: #{second_exception}"
+  end
+
+  if (!first_excepton.nil?) && second_exception.nil?
+    raise ScriptError, "only the whole-document parser threw: #{first_exception}"
```

Comment:
I don't the type matters too much, and this is just one that I've seen used in Ruby in situations that felt kind of similar to this.  I don't mind changing it but don't know what would be a better thing to change it to.